app/src/main/java/com/hadiyarajesh/notex/ui/note/add/AddNotesViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.hadiyarajesh.notex.ui.note.add
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hadiyarajesh.notex.database.dao.NoteDao
+import com.hadiyarajesh.notex.database.entity.Note
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+import java.time.Instant
+import javax.inject.Inject
+
+@HiltViewModel
+class AddNotesViewModel @Inject constructor(private val noteDao: NoteDao) : ViewModel() {
+    private val _eventFlow = MutableSharedFlow<UiEvent>()
+    val eventFlow = _eventFlow.asSharedFlow()
+
+    fun saveNote(noteState: NoteState, noteId: Long) {
+        Log.d("color_value", convertIntColorToHex(noteState.color.value))
+        viewModelScope.launch {
+
+            if (noteState.title.value.isNotBlank() && noteState.noteDesc.value.isNotBlank()) {
```

Comment:
Perform this UI validation in composable so we can directly show appropriate errors there

Reply:
okay